aws/resource_aws_qldb_stream.go

```diff
@@ -0,0 +1,355 @@
+package aws
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/qldb"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+// TODO:  Account for fact that streams can become "completed" and may impact future plans if this is not recognized as a valid state that does not need a re-apply to resolve...
+
+func resourceAwsQLDBStream() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsQLDBStreamCreate,
+		Read:   resourceAwsQLDBStreamRead,
+		Update: resourceAwsQLDBStreamUpdate,
+		Delete: resourceAwsQLDBStreamDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		// https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-qldb-stream.html
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"exclusive_end_time": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validateUTCTimestamp, // The ExclusiveEndTime must be in ISO 8601 date and time format and in Universal Coordinated Time (UTC). For example: 2019-06-13T21:36:34Z.
+				),
+			},
+			"inclusive_start_time": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validateUTCTimestamp,
+				),
+			},
+
+			"kinesis_configuration": {
```

Comment:
TLDR: I'm wondering the first vs the second here:

This would follow established patterns and allow for some re-use, but it won't let us enforce the requirement of a `stream_arn` value in a plan:
```
kinesis_configuration {
  aggregation_enabled = false
  stream_arn = ""
}
```

This would allow us to be more explicit (and require the `stream_arn` to be present, but it doesn't follow precedent:
```
kinesis_aggregation_enabled = false
kinesis_stream_arn                   = ""
```